crates/astria-celestia-client/src/blob_space.rs

```diff
@@ -84,12 +82,17 @@ impl RollupNamespaceData {
         &self,
         root_hash: [u8; 32],
     ) -> Result<(), RollupVerificationFailure> {
-        use sequencer_validation::MerkleTree;
-        let rollup_data_tree = MerkleTree::from_leaves(self.rollup_txs.clone());
+        let rollup_data_tree = merkle::Tree::from_leaves(&self.rollup_txs);
         let rollup_data_root = rollup_data_tree.root();
-        let mut leaf = self.chain_id.as_ref().to_vec();
-        leaf.append(&mut rollup_data_root.to_vec());
-        self.inclusion_proof.verify(&leaf, root_hash)?;
+        let mut leaf = Vec::with_capacity(self.chain_id.as_ref().len() + rollup_data_root.len());
+        leaf.extend_from_slice(self.chain_id.as_ref());
+        leaf.extend_from_slice(&rollup_data_root);
+        let reconstructed_root = self.inclusion_proof.reconstruct_root_with_leaf(&leaf);
+        if root_hash != reconstructed_root {
+            return Err(RollupVerificationFailure {
+                _private: (),
+            });
+        }
```

Comment:
i prefer the `inclusion_proof.verify(leaf, root_hash)` API, thoughts?

Reply:
yes, `fn(&self, leaf: &[u8], root: [u8; 32]) -> bool` seems good!

Reply:
I have renamed `Proof::is_leaf_in_tree` -> `Proof::verify`, and added the audit API with `Audit::perform`.